src/components/terminal.rs

```diff
@@ -0,0 +1,95 @@
+use crate::types::terminal_content::TerminalContent;
+use leptos::{ev::SubmitEvent, html::Input, *};
+use std::time::Duration;
+
+#[component]
+pub fn Terminal<F>(
+    cx: Scope,
+    // TODO - Tech Debt: Find a way to pass terminal in as mut
+    terminal: TerminalContent,
```

Comment:
Tried to pass it in as `mut` here, but I think there's a limitation with the Leptos component macro. Either way, this is an owned value, so you aren't getting a performance penalty by shadowing it later. (I think the idea is that you usually pass signals)